### PR TITLE
fix: check table existence in create table procedure

### PR DESCRIPTION
**Pull request by evenyag**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
This PR fixes that create table procedure doesn't check table existence before actually creating a region.

PR #1817 replaces the keys in the table engine and we should not use table's name to build the procedure's lock key.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/table-procedure/src/create.rs [hunk 1]
```diff
@@ -132,23 +132,22 @@ impl CreateTableProcedure {
...
-        {
-            logging::error!(
-                "Failed to create table {}, schema not found",
-                self.data.table_ref(),
-            );
-            return SchemaNotFoundSnafu {
-                name: &self.data.request.schema_name,
+            .context(AccessCatalogSnafu)?;
+        if table_exists && !self.data.request.create_if_not_exists {
+            return TableNotFoundSnafu {
```
Comment (WenyXu) [inline src/table-procedure/src/create.rs:145]: Should it be `TableAlreadyExist` instead?

File: src/table-procedure/src/create.rs [hunk 2]
```diff
@@ -132,23 +132,22 @@ impl CreateTableProcedure {
...
+                name: &self.data.request.table_name,
             }
             .fail()?;
+        } else if table_exists {
+            return Ok(Status::Done);
         }
```
Comment (WenyXu) [inline src/table-procedure/src/create.rs]:
``` suggestion
        if table_exists {
            return if self.data.request.create_if_not_exists {
                Ok(Status::Done)
            } else {
                TableExistsSnafu {
                    name: &self.data.request.table_name,
                }
                .fail()?
            };
        }
```

File: src/table-procedure/src/create.rs (hunk 1 above)
Comment (evenyag) [reply src/table-procedure/src/create.rs:145]: Oops 😢

**Review by WenyXu — APPROVED** [review]

File: src/table-procedure/src/error.rs [hunk 3]
```diff
@@ -72,10 +69,9 @@ impl ErrorExt for Error {
             }
             InvalidRawSchema { source, .. } => source.status_code(),
             AccessCatalog { source, .. } => source.status_code(),
-            CatalogNotFound { .. } | SchemaNotFound { .. } | TableExists { .. } => {
-                StatusCode::InvalidArguments
-            }
+            CatalogNotFound { .. } => StatusCode::InvalidArguments,
```
Comment (waynexia) [inline src/table-procedure/src/error.rs:72]: it looks like this is not used either

File: src/table-procedure/src/drop.rs [hunk 4]
```diff
@@ -122,18 +122,21 @@ impl DropTableProcedure {
...
-            .context(AccessCatalogSnafu)?
-            .context(TableNotFoundSnafu {
+            .context(AccessCatalogSnafu)?;
+        if !table_exists {
+            return TableNotFoundSnafu {
                 name: &request.table_name,
-            })?;
+            }
+            .fail()?;
+        }
```
Comment (waynexia) [inline src/table-procedure/src/drop.rs:139]: style: use `ensure!` instead?

**Review by waynexia — APPROVED** [review]
Not sure if this actually works 😵‍💫

**Pull request merged**